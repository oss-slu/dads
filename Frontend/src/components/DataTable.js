import * as React from 'react';


//component to render a dynamic tabel
//labels is list of strings for column titles
//data is a list of lists 
export default function DataTable({ labels, data }) {

    return (
        <>
            <table style={{ width: '500px' }}>
                <thead>
                    <tr>
                        {
                            labels.map((label) => <th>{label}</th>)
                        }
                    </tr>
                </thead>
                <tbody>
                    {
                        data.map((item, key) => <>
                            <tr style={{ textAlign: 'center' }}>
                                {
                                    item.map((element, id) =>
                                        <td>
                                            <span>
                                                {element}
                                            </span>
                                        </td>
                                    )
                                }
                            </tr>
                        </>)
                    }
                </tbody>
            </table>


        </>
    );
}