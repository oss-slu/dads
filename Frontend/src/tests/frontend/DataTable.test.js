// File: src/tests/frontend/tables/DataTable.test.js

import React from 'react';
import { render, screen } from '@testing-library/react';
import DataTable from '../../components/DataTable';
import '@testing-library/jest-dom';

describe('DataTable component', () => {
  const labels = ['Name', 'Age', 'Country'];
  const data = [
    ['Alice', 30, 'USA'],
    ['Bob', 25, 'Canada'],
    ['Charlie', 35, 'UK']
  ];

  test('renders without crashing', () => {
    render(<DataTable labels={labels} data={data} />);
    expect(screen.getByRole('table')).toBeInTheDocument();
  });

  test('renders all labels correctly', () => {
    render(<DataTable labels={labels} data={data} />);
    labels.forEach(label => {
      expect(screen.getByText(label)).toBeInTheDocument();
    });
  });

  test('renders all data rows correctly', () => {
    render(<DataTable labels={labels} data={data} />);
    data.forEach(row => {
      row.forEach(cell => {
        expect(screen.getByText(cell)).toBeInTheDocument();
      });
    });
  });

  test('renders pagination controls', () => {
    render(<DataTable labels={labels} data={data} />);
    expect(screen.getByText('First')).toBeInTheDocument();
    expect(screen.getByText('Prev')).toBeInTheDocument();
    expect(screen.getByText('Next')).toBeInTheDocument();
    expect(screen.getByText('Last')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<DataTable labels={labels} data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
