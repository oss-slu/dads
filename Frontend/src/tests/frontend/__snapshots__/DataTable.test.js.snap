// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable component matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="table table-sm table-bordered table-striped"
    id="resultstable"
    style="width: 700px;"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Age
        </th>
        <th>
          Country
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        style="text-align: center;"
      >
        <td>
          <span>
            Alice
          </span>
        </td>
        <td>
          <span>
            30
          </span>
        </td>
        <td>
          <span>
            USA
          </span>
        </td>
      </tr>
      <tr
        style="text-align: center;"
      >
        <td>
          <span>
            Bob
          </span>
        </td>
        <td>
          <span>
            25
          </span>
        </td>
        <td>
          <span>
            Canada
          </span>
        </td>
      </tr>
      <tr
        style="text-align: center;"
      >
        <td>
          <span>
            Charlie
          </span>
        </td>
        <td>
          <span>
            35
          </span>
        </td>
        <td>
          <span>
            UK
          </span>
        </td>
      </tr>
    </tbody>
  </table>
  <ul
    class="pagination pagination-lg"
  >
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
        <span
          class="visually-hidden"
        >
          First
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
        >
          ‹
        </span>
        <span
          class="visually-hidden"
        >
          Previous
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        1
      </a>
    </li>
    <li
      class="page-item active"
    >
      <span
        class="page-link"
      >
        12
        <span
          class="visually-hidden"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        20
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
        >
          ›
        </span>
        <span
          class="visually-hidden"
        >
          Next
        </span>
      </a>
    </li>
    <li
      class="page-item"
    >
      <a
        class="page-link"
        href="#"
        role="button"
        tabindex="0"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
        <span
          class="visually-hidden"
        >
          Last
        </span>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
